modules/geo/src/internalClusterTest/java/org/opensearch/geo/search/aggregations/bucket/GeoHashGridIT.java

```diff
@@ -268,14 +268,18 @@ public void testShardSizeIsZero() {
     }
 
     @Override
-    protected Set<String> generateBucketsForGeometry(final Geometry geometry, final GeoShapeDocValue geometryDocValue) {
+    protected Set<String> generateBucketsForGeometry(
+        final Geometry geometry,
+        final GeoShapeDocValue geometryDocValue,
+        boolean intersectingWithBB
+    ) {
         final GeoPoint topLeft = new GeoPoint();
         final GeoPoint bottomRight = new GeoPoint();
         assert geometry != null;
         GeoBoundsHelper.updateBoundsForGeometry(geometry, topLeft, bottomRight);
         final Set<String> geoHashes = new HashSet<>();
         for (int precision = MAX_PRECISION_FOR_GEO_SHAPES_AGG_TESTING; precision > 0; precision--) {
-            if (precision > 2 && !geometryDocValue.isIntersectingRectangle(boundingRectangleForGeoShapesAgg)) {
+            if (precision > MIN_PRECISION_WITHOUT_BB_AGGS && !intersectingWithBB) {
```

Comment:
nit. Try to avoid negatives... they can be easily missed...
```suggestion
            if (precision > MIN_PRECISION_WITHOUT_BB_AGGS && intersectingWithBB == false) {
```